pkg/kubelet/certificate/transport.go

```diff
@@ -79,7 +80,7 @@ func updateTransport(stopCh <-chan struct{}, period time.Duration, clientConfig
 	return d.CloseAll, nil
 }
 
-func addCertRotation(stopCh <-chan struct{}, period time.Duration, clientConfig *restclient.Config, clientCertificateManager certificate.Manager, exitAfter time.Duration, d *connrotation.Dialer) error {
+func addCertRotation(ctx context.Context, stopCh <-chan struct{}, period time.Duration, clientConfig *restclient.Config, clientCertificateManager certificate.Manager, exitAfter time.Duration, d *connrotation.Dialer) error {
```

Comment:
this has a stopCh, we must ensure that if we pass a context it actual cancels the function, as it is what people expect from it ... please avoid passing just the context to plumb down the logger, the context cancellation needs to happen